ibm/resource_ibm_network_gateway_test.go

```diff
@@ -0,0 +1,138 @@
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "hostname", hostname),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "domain", "terraformuat.ibm.com"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "os_reference_code", "OS_VYATTA_5600_5_X_UP_TO_1GBPS_SUBSCRIPTION_EDITION_64_BIT"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "datacenter", "ams01"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "network_speed", "100"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "hourly_billing", "false"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "private_network_only", "false"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "user_metadata", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "tcp_monitoring", "true"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "package_key_name", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "process_key_name", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "os_key_name", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "redundant_network", "false"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "public_bandwidth", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "memory", "{\"value\":\"newvalue\"}"),
+					resource.TestCheckResourceAttr(
+						"ibm_network_gateway.terraform-acceptance-test-1", "ipv6_enabled", "true"),
+					CheckStringSet(
+						"ibm_network_gateway.terraform-acceptance-test-1",
+						"tags", []string{"collectd"},
+					),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMNetworkGatewayConfig_basic(hostname string) string {
+	return fmt.Sprintf(`
+resource "ibm_network_gateway" "terraform-acceptance-test-1" {
+			hostname               = "%s"
+			domain                 = "terraformuat.ibm.com"
+			os_reference_code      = "OS_VYATTA_5600_5_X_UP_TO_1GBPS_SUBSCRIPTION_EDITION_64_BIT"
+			datacenter             = "ams01"
+			network_speed          = 100
+			hourly_billing         = true
+			private_network_only   = false
+			user_metadata          = "No metadata"
+			tcp_monitoring         = true
+			package_key_name       = "NETWORK_GATEWAY_APPLIANCE"
```

Comment:
The package name should be constant hard-coded value and doesn't need to be configurable by the user

Reply:
@ashishth09 thanks for all the feedback and comments, I have modified the code to include all the changes.

Please share your thoughts on it.